sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -553,7 +554,10 @@ export class EventHubSender extends LinkEntity {
           this._sender!.on(SenderEvents.rejected, onRejected);
           this._sender!.on(SenderEvents.modified, onModified);
           this._sender!.on(SenderEvents.released, onReleased);
-          waitTimer = setTimeout(actionAfterTimeout, Constants.defaultOperationTimeoutInSeconds * 1000);
+          waitTimer = setTimeout(
+            actionAfterTimeout,
+            options && options.idleTimeout ? options.idleTimeout : Constants.defaultOperationTimeoutInSeconds * 1000
```

Comment:
`options && options.idleTimeout` is "falsy" if the timeout was set to `0`
Use the user provided value only if it is greater than 0.